week3/homework/src/actions/markAsNotDone.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const deserializeTodo = require('./deserializeTodo');
+
+function markAsNotDone(todo, request, response) {
+  deserializeTodo(request, response)
+    .then(() => {
+      const id = request.params.id;
+      return todo.notDone(id);
+    })
+    .then(todo => {
+      response.status(200);
+      response.json({ todo });
+    })
+    .catch(({ message, code }) => {
+      response.status(code === 'not-found' ? 404 : 500);
```

Comment:
this code is also repeating many times. Can you think of a way to eliminate the duplication?

Reply:
Ofcourse, I can think about it.